File: Project Euler/p_25.py
Fib = []

Fib.append(0)
Fib.append(1)

for i in range(2, 5000):
    Fib.append(Fib[i - 1] + Fib[i - 2])

for i in range(4000, 5000):
    if (len(str(Fib[i])) >= 1000):
        print(i)
        
        break

File: Project Euler/p_25.java
import java.util.*;
import java.io.*;
import java.math.*;

public class p_25 {
    public static BigInteger fib (int n) {
        BigInteger a = BigInteger.valueOf(0);
        BigInteger b = BigInteger.valueOf(1);
        BigInteger c = BigInteger.valueOf(1);

        for (int i = 2; i < n; i++) {
            c = a.add(b);

            a = b;
            b = c;
        }

        return c;
    }

    public static void main(String[] args) {
        for (int i = 4000; i < 5000; i++) {
            if (fib(i).toString().length() >= 1000) {
                System.out.println(i);

                break;
            }
        }
    }
}